refactor(transformer/jsx): do not take `None` when extracting `key` prop

overlookmotel opened the pull request:
Tiny refactor to JSX transform. There's no point setting `key_prop = None`, so skip this branch unless the `key` attribute has a value (it always should).

We can also remove the `.take()` call.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9228?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#9228** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9228?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9228?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#9227** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9227?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9226** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9226?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9228).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented:
> ❌	`semantic[RadixUIAdoptionSection.jsx]`	72.2 µs	74.4 µs	-3.08%

Oh man! Not this nonsense again... this benchmark result is clearly erroneous noise. This PR doesn't touch Semantic!

Dunqing commented:
### Merge activity

* **Feb 19, 9:45 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(transformer/jsx): do not take `None` when extracting `key` prop (#9228)

Tiny refactor to JSX transform. There's no point setting `key_prop = None`, so skip this branch unless the `key` attribute has a value (it always should).

We can also remove the `.take()` call.
